package.json

```diff
@@ -28,7 +28,7 @@
     "eslint": "^8.27.0",
     "prettier": "^2.1.1",
     "release-it": "^15.5.0",
-    "typescript": "~4.8.0"
+    "typescript": "~5.2.2"
```

Comment:
why upgrade TS? 
it looks like it broke the lints :sweat_smile:

Reply:
The setting to support importing files with typescript extensions ("allowImportingTsExtensions") was added in Typescript 5. Perhaps it would be better to leave the TS version at 4.8 anyway? This is the feedback I'm hoping for on this PR.

Reply:
is it possible to support `allowImportingTsExtensions` _and_ TS 4.8?
Like, have some version check somewhere?

I think the only way Glint could have a breaking change in `@glint/core` is if it broke lockstep with all the other packages -- introducing a breaking change in the other packages would have an obnoxious impact on the ecosystem, imo

Reply:
The other changes only kick in if you have an import with a .gts extension, so I guess it is probably OK to not require 4.8.0 here but document that you need TS 5 if you want to import with extensions?

Reply:
With Typescript 4.8.0. there is a failing test:

```
src/index.gts:1:1 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './Greeting.js' instead.
```

I think we need TS5 and `allowImportingTsExtensions: true` in order for this to pass. I'll look into whether I can use TS5 for just that test.

Reply:
we can also conditionally skip the test, and be sure to add TS 4.8 to the test matrix, as well as TS 5.x

Reply:
I think I figured it out. Let's see how this CI run does.

Reply:
🤣 failed comically quickly! I'll pick this up again later.

Reply:
I do believe this is resolved now